### PR TITLE
Add AJAX request error handling for instructorFeedbackEdit.js #4386 

### LINKED ISSUE #4386: Add AJAX request error handling for instructorFeedbackEdit.js
Part of #4082 

**instructorFeedbackEdit.js**

`getVisibilityMessage()`

**Success:** Gets messages for the visibility of a question from the server and replaces the current visibility messages. Shows the messages and hide the options.
**Failure:** Hides the option, shows nothing in return. Upon hitting an error, the messages will not be displayed upon retrying unless the options are changed (due to a check to prevent duplicate requests).

The preview visibility label should be shown with an error message.

### PR DESCRIPTION
**Pull request by yj-soh**
Fixes #4386

### CODE REVIEW

**Comment by kanghj** [comment]
After showing the failure message, when the user clicks on the button and it succeeds, the error message still stays on the button. If the ajax request succeeds, should we remove the message asking them to retry?

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 1]
```diff
@@ -820,12 +818,22 @@ function getVisibilityMessage(buttonElem) {
...
+            previousFormDataMap[qnNumber] = formData;
+            
             $formVisibility.html(formatVisibilityMessageHtml(data.visibilityMessage));
             $formVisibility.show();
             $formOptions.hide();
         },
         error: function(jqXHR, textStatus, errorThrown) {
-
+            var visibilityButton = $form.find('.visibilityMessageButton');
+            var radioInput = visibilityButton.children().first();
```
Comment (kanghj) [inline src/main/webapp/js/instructorFeedbackEdit.js]: I am guessing this code is needed, but I don't know why? Leave a comment in the code, if needed. 

Using children.first might not be the best way to get a radio input, might be better if you just explicitly select the radio input, if the radio input is what you need here.

**Comment by kanghj** [comment]
How about adding a test too?

**Comment by yj-soh** [comment]
Do we have any standard manner of testing AJAX failure conditions?
I tried a quick search, but it doesn't seem possible without using some sort of JS testing library.

**Comment by LowWeiLin** [comment]
There are some places where we test ajax failure by injecting an invalid URL for the request. Can see #4279 for reference.

**Comment by kanghj** [comment]
Fix this soon?

**Comment by yj-soh** [comment]
@kanghj Test added.

**Comment by kanghj** [comment]
Looks good

**Pull request merged**